### PR TITLE
Migrate dav application from ILogger to LoggerInterface

### PR DESCRIPTION
**Pull request by come-nc**
Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

### CODE REVIEW

File: apps/dav/lib/AppInfo/Application.php [hunk 1]
```diff
@@ -323,7 +323,7 @@ function (GenericEvent $event) use ($container, $serverContainer) {
 				$job->run([]);
 				$serverContainer->getJobList()->setLastRun($job);
 			} catch (Exception $ex) {
-				$serverContainer->getLogger()->logException($ex);
+				$serverContainer->get(LoggerInterface::class)->error($ex->getMessage(), ['exception' => $e]);
```
Comment (tcitworld) [inline apps/dav/lib/AppInfo/Application.php:326]:
```suggestion
				$serverContainer->get(LoggerInterface::class)->error($ex->getMessage(), ['exception' => $ex]);
```

File: apps/dav/lib/Connector/Sabre/Auth.php [hunk 2]
```diff
@@ -157,7 +157,7 @@ public function check(RequestInterface $request, ResponseInterface $response) {
 		} catch (Exception $e) {
 			$class = get_class($e);
 			$msg = $e->getMessage();
-			\OC::$server->getLogger()->logException($e);
+			\OC::$server->get(LoggerInterface::class)->error($e->getMessage(), ['exception' => $e]);
```
Comment (tcitworld) [inline apps/dav/lib/Connector/Sabre/Auth.php:160]: Lacks an ` use Psr\Log\LoggerInterface;`

File: apps/dav/tests/integration/UserMigration/CalendarMigratorTest.php [hunk 3]
```diff
@@ -85,7 +85,7 @@ private function getProperties(VCalendar $vCalendar): array {
 			fn (VObjectProperty $property) => $property->serialize(),
 			array_values(array_filter(
 				$vCalendar->children(),
-				fn (mixed $child) => $child instanceof VObjectProperty,
```
Comment (tcitworld) [inline apps/dav/tests/integration/UserMigration/CalendarMigratorTest.php:88]: Can you make this change in another commit?

File: apps/dav/tests/unit/CalDAV/Publishing/PublishingTest.php [hunk 4]
```diff
@@ -36,7 +36,7 @@
 use Sabre\HTTP\Response;
 use Test\TestCase;
 
-class PluginTest extends TestCase {
+class PublishingTest extends TestCase {
```
Comment (tcitworld) [inline apps/dav/tests/unit/CalDAV/Publishing/PublishingTest.php:39]: Can you make this change in another commit?

File: apps/dav/tests/unit/CalDAV/WebcalCaching/RefreshWebcalServiceTest.php [hunk 5]
```diff
@@ -24,7 +24,7 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  *
  */
-namespace OCA\DAV\Tests\unit\BackgroundJob;
+namespace OCA\DAV\Tests\unit\CalDAV\WebcalCaching;
```
Comment (tcitworld) [inline apps/dav/tests/unit/CalDAV/WebcalCaching/RefreshWebcalServiceTest.php:27]: Can you make this change in another commit?

**Review by tcitworld — COMMENTED** [review]
Looks real good overall !

**Comment by come-nc** [comment]
Rebased on master

**Review by CarlSchwan — APPROVED** [review]

**Comment by come-nc** [comment]
@tcitworld @ChristophWurst @miaulalala Can one of you review this? The drone failure is unrelated and now that stable24 is branched out this is good to be merged in master.

**Comment by come-nc** [comment]
Rebased on master

**Review by tcitworld — APPROVED** [review]
